### PR TITLE
Mark test cases end-to-end for Rocket team components

### PR DESCRIPTION
**Pull request by shweta83**
Component | Status
Ansible - Done
Provisioning Templates - Done
Puppet - Done
Registration - Done
Discovery Image - Done
Discovery Plugin - Done
Provisioning - Done
Convert2Rhel - Done 
Compute Resources - CNV - No automated test case 
Compute Resources - VMWare - Pending(No test has end-to-end coverage)
Compute Resources - Libvirt - Pending(No test has end-to-end coverage)
Compute Resources - EC2 - Pending(No test has end-to-end coverage)
Compute Resources -RHEV - Done
Compute Resources -GCE - Done
Compute Resources -OpenStack - Pending (CRUD test case marked, No test has end-to-end coverage)
Compute Resources -Azure - Done
SCAP Plugin - Done
Bootdisk Plugin - Pending(No test has end-to-end coverage)
DHCPDNS - Pending(No test has end-to-end coverage)
Host Form - Not needed
Infoblox integration - Pending(No test has end-to-end coverage)
Notifications - Pending(No test has end-to-end coverage)
Compute Resources - Pending(No test has end-to-end coverage)
Fact - Pending(No test has end-to-end coverage)
Logging - Done
Networking - Done
Parameters - Pending(No test has end-to-end coverage)
Settings - Pending(No test has end-to-end coverage)
TFTP - Pending(No test has end-to-end coverage)
Pulp - Not needed
Candlepin - Not needed

### CODE REVIEW

**Comment by shweta83** [comment]
> Hey @shweta83 can we mark Azure CLI test cases as well `test_positive_crud_azurerm_cr` `test_positive_azurerm_host_provisioned`

Added first one as discussed which covers crud operations for azure cr.

**Review by adarshdubey-star — APPROVED** [review]

File: tests/foreman/cli/test_logging.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-"""CLI tests for logging.
+"""CLI tests for pytestmark = [pytest.mark.no_containers, pytest.mark.destructive]logging.
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_logging.py:1]: Looks unintended 
```suggestion
"""CLI tests for logging.
```

**Review by shubhamsg199 — COMMENTED** [review]
@shweta83 Can we also add e2e marker for this Computeresource rhev test ?https://github.com/SatelliteQE/robottelo/blob/d5753445bb8bbeddad5d97e03e1be08ddc551687/tests/foreman/cli/test_computeresource_rhev.py#L382

File: tests/foreman/cli/test_logging.py (hunk 1 above)
Comment (shweta83) [reply tests/foreman/cli/test_logging.py]: Yes. It was done by mistake.

**Comment by shweta83** [comment]
> @shweta83 Can we also add e2e marker for this Computeresource rhev test ?
> 
> https://github.com/SatelliteQE/robottelo/blob/d5753445bb8bbeddad5d97e03e1be08ddc551687/tests/foreman/cli/test_computeresource_rhev.py#L382

It doesn't cover end-to-end so skipped it.

**Comment by shubhamsg199** [comment]
> > @shweta83 Can we also add e2e marker for this Computeresource rhev test ?
> > https://github.com/SatelliteQE/robottelo/blob/d5753445bb8bbeddad5d97e03e1be08ddc551687/tests/foreman/cli/test_computeresource_rhev.py#L382
> 
> It doesn't cover end-to-end so skipped it.

Well, it actually does cover end-to-end functionality of creating a CR and provisioning a host with it, which is what we expect :)

**Comment by shweta83** [comment]
> > > @shweta83 Can we also add e2e marker for this Computeresource rhev test ?
> > > https://github.com/SatelliteQE/robottelo/blob/d5753445bb8bbeddad5d97e03e1be08ddc551687/tests/foreman/cli/test_computeresource_rhev.py#L382
> > 
> > 
> > It doesn't cover end-to-end so skipped it.
> 
> Well, it actually does cover end-to-end functionality of creating a CR and provisioning a host with it, which is what we expect :)

There is another test case with same name in longrun. I was confused with that. Updated the marker.

**Review by shubhamsg199 — APPROVED** [review]
ACK

**Review by Gauravtalreja1 — APPROVED** [review]

File: tests/foreman/api/test_computeresource_azurerm.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 class TestAzureRMComputeResourceTestCase:
     """Tests for ``api/v2/compute_resources``"""
 
+    @pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_computeresource_azurerm.py:34]: A CRUD (Create, Read, Update, Delete) test by itself is not an end-to-end test, as it typically only focuses on testing the individual functionalities of creating, reading, updating, and deleting data within the application. However, a CRUD test could be a part of an end-to-end test, which would involve testing the entire workflow of a user interacting with the application and performing the CRUD operations as part of that workflow.

e.g. `Compute Resource` Create, Update, and Delete do not make any value for the customer as end to end. They are not actually using the any feature. But compute resource used for host provisioning make as end to end test. Think like a customer here, will any cusotmer will just create Compute Resource and update and delete it? Customer will use compute resource with host provisoning then it make sense as end to end test. 

Please let me know yout thoughts here ?

File: tests/foreman/api/test_computeresource_azurerm.py [hunk 3]
```diff
@@ -31,6 +31,7 @@
...
     @pytest.mark.upgrade
     @pytest.mark.tier1
     @pytest.mark.parametrize(
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_computeresource_azurerm.py:37]: `test_positive_azurerm_host_provisioned` is more an ideal End to End Test, As it is testing whole functionality with Compute Resource.

File: tests/foreman/api/test_computeresource_gce.py [hunk 4]
```diff
@@ -164,6 +164,7 @@ def gce_hostgroup_puppet(
 class TestGCEComputeResourceTestCases:
     """Tests for ``api/v2/compute_resources``."""
 
+    @pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_computeresource_gce.py:167]: Same here as I mentioned in my first statement, It is not an ideal end-to-end test. This is just a CRUD test.

File: tests/foreman/cli/test_computeresource_azurerm.py [hunk 5]
```diff
@@ -57,6 +57,7 @@ def azurerm_hostgroup(
 class TestAzureRMComputeResourceTestCase:
     """AzureRm compute resource Tests"""
 
+    @pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_computeresource_azurerm.py:60]: Invalid end-to-end test.

File: tests/foreman/cli/test_computeresource_osp.py [hunk 6]
```diff
@@ -45,6 +45,7 @@ def cr_cleanup(self, cr_id):
         except CLIReturnCodeError:
             pass
 
+    @pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_computeresource_osp.py:48]: invalid end to end test

File: tests/foreman/cli/test_logging.py [hunk 7]
```diff
@@ -29,6 +29,8 @@
 from robottelo.config import settings
 from robottelo.logging import logger
 
+pytestmark = pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_logging.py:32]: I don't feel these are end to end tests

File: tests/foreman/cli/test_subnet.py [hunk 8]
```diff
@@ -72,6 +72,7 @@ def invalid_missing_attributes():
     ]
 
 
+@pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_subnet.py:75]: invalid end to end test

**Review by omkarkhatavkar — CHANGES_REQUESTED** [review]
I personally think CRUD tests can not be End to End Tests, Although, a CRUD test could be a part of an end-to-end test, which would involve testing the entire workflow of a user interacting with the application and performing the CRUD operations as part of that workflow.

File: tests/foreman/api/test_computeresource_azurerm.py (hunk 2 above)
Comment (shweta83) [reply tests/foreman/api/test_computeresource_azurerm.py:34]: Agree. Updated the test markers for the compute resources components.

File: tests/foreman/api/test_computeresource_azurerm.py (hunk 3 above)
Comment (shweta83) [reply tests/foreman/api/test_computeresource_azurerm.py:37]: Updated.

File: tests/foreman/api/test_computeresource_gce.py (hunk 4 above)
Comment (shweta83) [reply tests/foreman/api/test_computeresource_gce.py:167]: Updated.

File: tests/foreman/cli/test_logging.py (hunk 7 above)
Comment (shweta83) [reply tests/foreman/cli/test_logging.py:32]: These together marks the logging of different components.

File: tests/foreman/api/test_computeresource_azurerm.py (hunk 2 above)
Comment (omkarkhatavkar) [reply tests/foreman/api/test_computeresource_azurerm.py:34]: We can remove this as e2e.

File: tests/foreman/api/test_classparameters.py [hunk 9]
```diff
@@ -388,6 +392,7 @@ def test_negative_validate_matcher_and_default_value(
             in context.value.response.text
         )
 
+    @pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_classparameters.py:395]: these are validation tests can not be a e2e tests

File: tests/foreman/api/test_classparameters.py [hunk 10]
```diff
@@ -331,6 +334,7 @@ def test_positive_validate_matcher_value_with_list(
         sc_param.update(['override', 'default_value', 'validator_type', 'validator_rule'])
         assert sc_param.read().default_value == 'example'
 
+    @pytest.mark.e2e
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_classparameters.py:337]: these are validation tests that can not be an e2e tests

File: tests/foreman/api/test_computeresource_gce.py (hunk 4 above)
Comment (omkarkhatavkar) [reply tests/foreman/api/test_computeresource_gce.py:167]: remove marker from this test

File: tests/foreman/cli/test_computeresource_azurerm.py (hunk 5 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_computeresource_azurerm.py:60]: remove 'e2e' marker

File: tests/foreman/cli/test_logging.py (hunk 7 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_logging.py:32]: Okay I think for this component we can make this as e2e.

File: tests/foreman/cli/test_subnet.py (hunk 8 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_subnet.py:75]: remove the marker here `e2e`

**Review by omkarkhatavkar — CHANGES_REQUESTED** [review]
Minor changes are Missing

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Pull request merged**